app/src/common/chromium/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java

```diff
@@ -0,0 +1,333 @@
+        public class Choice implements WSession.PromptDelegate.ChoicePrompt.Choice {
+            public final boolean disabled;
+            public final @Nullable String icon;
+            public final @NonNull String id;
+            public final @NonNull String label;
+            public boolean selected;
+            public final boolean separator;
+            public @Nullable Choice[] choices;
+
+            /* package */ Choice(SelectPopupItem item, int id, boolean selected) {
+                this.disabled = !item.isEnabled();
+                this.id =  String.valueOf(id);
+                this.icon = null;
+                this.label = item.getLabel();
+                this.separator = false;
+                this.selected = selected;
+                this.choices = null;
+            }
+
+            @Override
+            public boolean disabled() {
+                return this.disabled;
+            }
+
+            @Nullable
+            @Override
+            public String icon() {
+                return this.icon;
+            }
+
+            @NonNull
+            @Override
+            public String id() {
+                return this.id;
+            }
+
+            @NonNull
+            @Override
+            public String label() {
+                return this.label;
+            }
+
+            @Override
+            public boolean selected() {
+                return this.selected;
+            }
+
+            @Override
+            public boolean separator() {
+                return this.separator;
+            }
+
+            @Override
+            @Nullable
+            public Choice[] items() { return this.choices; }
+
+        }
+        public final @Nullable String message;
+
+        public final @WSession.PromptDelegate.ChoicePrompt.ChoiceType int type;
+
+        public final @Nullable Choice[] choices;
+
+        private final Callback<int[]> mSelectionChangedCallback;
+
+        private Choice createChoice(List<SelectPopupItem> items, int[] position, int[] selected,
+                                    int[] selectedIndex) {
+            if (items.size() == position[0])
+                return null;
+
+            SelectPopupItem item = items.get(position[0]);
+
+            boolean isSelected = false;
+            if (selected.length > selectedIndex[0] && selected[selectedIndex[0]] == position[0]) {
+                isSelected = true;
+                selectedIndex[0]++;
+            }
+
+            Choice choice = new Choice(item, position[0], isSelected);
+            if (item.isGroupHeader()) {
+                ArrayList<Choice> choices = new ArrayList<>();
+                while (items.size() > position[0]+1) {
+                    SelectPopupItem child = items.get(position[0]+1);
+                    if (child.isGroupHeader())
+                        break;
+
+                    position[0] += 1;
+                    choices.add(createChoice(items, position, selected, selectedIndex));
```

Comment:
`createChoice` should not return null, since we already make sure of the position before calling `createChoice`. I think we can put here assert if the return value is null.

Reply:
added `assert child != null`